index.js

```diff
@@ -6,11 +6,32 @@
 // https://github.com/ethereum/web3.js
 /////////////////////////////////////////////////////
 
-import AWS from 'aws-sdk';
-import HttpProvider from 'web3-providers-http';
-import XHR2 from 'xhr2';
+const AWS = require('aws-sdk');
+const HttpProvider = require('web3-providers-http');
+const XHR2 = require('xhr2');
+
+const validateCredentials = (credentials) => {
+  let valid = true;
+
+  //ensure the provided object has property accessKeyId
+  if (!credentials.hasOwnProperty('accessKeyId')) valid = false
+  //ensure the provided object has property secretAccessKey
+  if (!credentials.hasOwnProperty('secretAccessKey')) valid = false
+  //ensure accessKeyId is not undefined or empty
+  if (credentials.accessKeyId == undefined || credentials.accessKeyId === "") valid = false
+  //ensure secretAccessKey is not undefined or empty
+  if (credentials.secretAccessKey == undefined || credentials.secretAccessKey === "") valid = false
+  
+
+  return valid;
+}
+
+module.exports = class AWSHttpProvider extends HttpProvider {
+  constructor(host, ssmCredentials) {
+      super(host)
```

Comment:
Please keep indentation level consistent (2 chars per indent) and add semicolon at the end of this line.

Reply:
resolved